bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/AbstractZkLedgerManager.java

```diff
@@ -409,7 +426,7 @@ public void processResult(int rc, String path, Object ctx, byte[] data, Stat sta
     public void writeLedgerMetadata(final long ledgerId, final LedgerMetadata metadata,
                                     final GenericCallback<Void> cb) {
         Version v = metadata.getVersion();
-        if (Version.NEW == v || !(v instanceof LongVersion)) {
+        if (!(v instanceof LongVersion)) {
```

Comment:
Why are we removing the first check? the (v instanceof LongVersion) just checks the instance, not the actual compare right?

Reply:
this is a bug. if you take a look at this line (https://github.com/apache/bookkeeper/blob/master/bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/AbstractZkLedgerManager.java#L416), it directly casts v to LongVersion. so the passed-in `metadata.getVersion()` has to be `LongVersion`.

Reply:
@sijie do you mean that it is now possible for *v* to be '==' to Version.NEW because it cannot be of the same class so that the test is not useful?

Reply:
@eolivelli I mean v can not be `Version.New` because line 416 expects a LongVersion. also the ledger manager api have separate api for CreateLedgerMetadata and WriteLedgerMetadata. so we don't expect Version.New and Version.ANY on `WriteLedgerMetadata`.